dbt/adapters/snowflake/connections.py

```diff
@@ -218,38 +218,48 @@ def open(cls, connection):
             logger.debug('Connection is already open, skipping open.')
             return connection
 
-        try:
-            creds = connection.credentials
-
-            handle = snowflake.connector.connect(
-                account=creds.account,
-                user=creds.user,
-                database=creds.database,
-                schema=creds.schema,
-                warehouse=creds.warehouse,
-                role=creds.role,
-                autocommit=True,
-                client_session_keep_alive=creds.client_session_keep_alive,
-                application='dbt',
-                **creds.auth_args()
-            )
-
-            if creds.query_tag:
-                handle.cursor().execute(
-                    ("alter session set query_tag = '{}'")
-                    .format(creds.query_tag))
-
-            connection.handle = handle
-            connection.state = 'open'
-        except snowflake.connector.errors.Error as e:
+        error = None
+        for attempt in range(1, 4):  # retry opening the connection twice after encountering an error
+            try:
+                creds = connection.credentials
+
+                handle = snowflake.connector.connect(
+                    account=creds.account,
+                    user=creds.user,
+                    database=creds.database,
+                    schema=creds.schema,
+                    warehouse=creds.warehouse,
+                    role=creds.role,
+                    autocommit=True,
+                    client_session_keep_alive=creds.client_session_keep_alive,
+                    application='dbt',
+                    **creds.auth_args()
+                )
+
+                if creds.query_tag:
+                    handle.cursor().execute(
+                        ("alter session set query_tag = '{}'")
+                            .format(creds.query_tag))
+
+                connection.handle = handle
+                connection.state = 'open'
+            except snowflake.connector.errors.Error as e:
```

Comment:
I'm glad we're catching an exception as defined by `snowflake.connector`. It looks like [they define](https://github.com/snowflakedb/snowflake-connector-python/blob/master/src/snowflake/connector/errors.py) some more specific error codes for retry. Should we try to be more precise here, and check whether the error code is retryable? Or better to always retry on all connections? I imagine the latter is better, just figured it's worth double-checking.